examples/api/lib/widgets/implicit_animations/animated_value.0.dart

```diff
@@ -0,0 +1,371 @@
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for [AnimatedValue].
+
+void main() => runApp(const AnimatedValueApp());
+
+/// Prizes come in all shapes!
+enum Prize implements ValueKey<Prize> {
+  circle,
+  triangle,
+  square,
+  hexagon;
+
+  @override
+  Prize get value => this;
+
+  @override
+  String toString() => name;
+}
+
+class PrizeSpinner extends StatelessWidget {
+  const PrizeSpinner({super.key});
+
+  // This build method returns an AnimatedRotation, which is a subtype
+  // of AnimatedValue!
+  @override
+  AnimatedValue<double> build(BuildContext context) {
+    final Prize? prize = FreePrizes.maybeOf(context);
+    final double turns = (prize?.index ?? math.Random().nextDouble()) / 4 + 1 / 8;
+
+    // Changing a widget's key will cause its state to reset.
+    // So if an AnimatedValue widget animates from an initialValue,
+    // changing the key will make the animation run again!
+
+    return AnimatedRotation(
+      key: prize,
+      initialValue: prize != null ? turns - 2 : null,
+      turns: turns,
+      duration: const Duration(seconds: 2),
+      curve: Curves.easeOutCubic,
+      onEnd: () {
+        if (prize != null) {
+          FreePrizes.recolor(context);
+
+          showDialog<void>(
+            context: context,
+            builder: (BuildContext context) => PrizePopup(prize: prize),
+          );
+        }
+      },
+      child: const Align(
+        alignment: Alignment.centerRight,
+        child: Icon(
+          Icons.arrow_right_alt_rounded,
+          size: 125,
+          color: Color(0xff202020),
+        ),
+      ),
+    );
+  }
+}
+
+abstract interface class _PrizeScope {
+  /// Randomize the prize!
+  void randomize();
+
+  /// Color the screen!
+  void recolor();
+}
+
+class FreePrizes extends InheritedWidget {
+  const FreePrizes({super.key, this.prize, required super.child});
+
+  final Prize? prize;
+
+  static Prize? maybeOf(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType<FreePrizes>()?.prize;
+  }
+
+  static _PrizeScope _prizeScope(BuildContext context) {
+    return context.findAncestorStateOfType<_AnimatedValueAppState>()!;
+  }
+
+  static void spin(BuildContext context) {
+    _prizeScope(context).randomize();
+  }
+
+  static void recolor(BuildContext context) {
+    _prizeScope(context).recolor();
+  }
+
+  @override
+  bool updateShouldNotify(FreePrizes oldWidget) => prize != oldWidget.prize;
+}
+
+class AnimatedValueApp extends StatefulWidget {
+  const AnimatedValueApp({super.key});
+
+  @override
+  State<AnimatedValueApp> createState() => _AnimatedValueAppState();
+}
+
+class _AnimatedValueAppState extends State<AnimatedValueApp> implements _PrizeScope {
+  Prize? prize;
+
+  static final math.Random rng = math.Random();
+  @override
+  void randomize() {
+    final int newIndex = switch (prize?.index) {
+      final int index => index + 1 + rng.nextInt(3),
+      null => rng.nextInt(4),
+    };
```

Comment:
Could be simplified to a simple if/else, no?

Reply:
Also, conceptually I am trying to understand why the newIndex depends on the current index?